src/syft/core/nodes/abstract/remote_nodes.py

```diff
@@ -0,0 +1,94 @@
+import enum
+from .client import Client
+from ...io.route import Route
+from ...message.syft_message import SyftMessage
+
+class NodeTypes(enum):
+    Network = 1
+    Domain = 2
+    Device = 3
+    VM = 4
+
+class Node(object):
+    def __init__(self, type, id = None, name = None, route = None, tags=[]):
+        self.id = id
+        self.name = name
+        self.type = type
+        self.route = route
```

Comment:
Oh interesting - I'm starting to see how you're thinking about "Route" here. I think of "Route" as only being something messages have - but if you're also thinking that Nodes would hold a "Route" object to know how to communicate (which totally makes sense) then I see why you would want to have communication information (like host and port) included.

Very interesting.